compiler/Core/CompilerState.h

```diff
@@ -42,13 +42,31 @@ struct IDTable {
     void defineGlobalVariables(llvm::LLVMContext &lctx, llvm::Module &module, llvm::IRBuilderBase &builder);
 };
 
+struct RubyStringTable {
+    struct RubyStringTableEntry {
```

Comment:
These structures should all really be `Entry`, since they are scoped to the table class they are a part of.  I plan on making that change in a followup to this PR.